src/Compilers/VisualBasic/Portable/CodeGen/EmitExpression.vb

```diff
@@ -288,35 +288,43 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeGen
                 Dim nullCheckOnCopy = conditional.CaptureReceiver OrElse (receiverType.IsReferenceType AndAlso receiverType.TypeKind = TypeKind.TypeParameter)
 
                 If nullCheckOnCopy Then
-                    EmitReceiverRef(receiver, isAccessConstrained:=Not receiverType.IsReferenceType, addressKind:=AddressKind.ReadOnly)
+                    Dim tempLocal As LocalDefinition = EmitReceiverRef(receiver, isAccessConstrained:=Not receiverType.IsReferenceType, addressKind:=AddressKind.ReadOnly)
```

Comment:
We never freed this temp previously, which is why I'm not assigning this to recieverTemp (which gets freed later) and instead I'm assigning it to a new variable.

However this seems wrong to me. Could someone who understands codeGen better than I do confirm?

Reply:
It looks like this was a codegen improvement that was made to C# and not to VB. I suspect there was no particular reason, we just weren't as concerned with codegen quality in VB.

If you want to, you could use the `receiverTemp` and add the the check to free it. Alternatively, this is fine, although I'd leave a comment about why you're doing it.